File: sentil-java/src/main/java/io/github/sedislab/sentil/BayesConfig.java
package io.github.sedislab.sentil;

/** Bayesian sequential test settings. */
public final class BayesConfig {
    double threshold;
    double bayesFactor = 100.0;
    long maxSamples = 100000;
    long seed = 42;

    public BayesConfig(double threshold) {
        this.threshold = threshold;
    }

    public double threshold() {
        return threshold;
    }

    public double bayesFactor() {
        return bayesFactor;
    }

    public BayesConfig bayesFactor(double bayesFactor) {
        this.bayesFactor = bayesFactor;
        return this;
    }

    public long maxSamples() {
        return maxSamples;
    }

    public BayesConfig maxSamples(long maxSamples) {
        this.maxSamples = maxSamples;
        return this;
    }

    public long seed() {
        return seed;
    }

    public BayesConfig seed(long seed) {
        this.seed = seed;
        return this;
    }
}

File: sentil-matlab/+sentil/BayesConfig.m
classdef BayesConfig
    %BAYESCONFIG Settings for a Bayesian sequential test.

    properties
        threshold (1, 1) double
        bayes_factor (1, 1) double = 100
        max_samples (1, 1) double = 100000
        seed (1, 1) double = 42
    end

    methods
        function obj = BayesConfig(threshold)
            %BAYESCONFIG A config for the asserted probability threshold.
            if nargin >= 1
                obj.threshold = threshold;
            end
        end
    end
end